.circleci/config.yml

```diff
@@ -79,36 +98,69 @@ jobs:
             cd ansible && ansible-playbook -e 'ansible_python_interpreter="/usr/bin/env python"' \
               --check -i localhost, environments.yml
 
-  deploy:
+  setup_uat:
     <<: *container_config
+    environment:
+      - DEPLOY_ENV: uat
     steps:
       - checkout
       - *restore_venv
       - *activate_venv
       - *restore_roles
-      # Don't run on UAT until we're sure it won't cause changes
-      - run:
-          name: Setup infrastructure
-          command: |
-            cd ansible && ansible-playbook -e 'ansible_python_interpreter="/usr/bin/env python"' \
-              --skip-tags "uat-env" environments.yml
-      - run:
-          name: Setup environments
-          command: |
-            cd ansible && ansible-playbook --limit='tag_role_epoch' setup.yml
-      # split monitoring playbook by environment because Ansible group vars are merged otherwise
-      - run:
-          name: Install monitoring tools (Fast Integration)
-          command: |
-            cd ansible && ansible-playbook --limit='tag_env_fast_integration:&tag_role_epoch' monitoring.yml
-      - run:
-          name: Install monitoring tools (Integration)
-          command: |
-            cd ansible && ansible-playbook --limit='tag_env_integration:&tag_role_epoch' monitoring.yml
-      - run:
-          name: Install monitoring tools (UAT)
-          command: |
-            cd ansible && ansible-playbook --limit='tag_env_uat:&tag_role_epoch' monitoring.yml
+      # Don't run infrastructure setup on UAT until we're sure it won't cause changes
+      # - *setup_infra
+      - *setup_node
+      - *setup_monitoring
+
+  setup_integration:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: integration
+    steps:
+      - checkout
+      - *restore_venv
+      - *activate_venv
+      - *restore_roles
+      - *setup_infra
+      - *setup_node
+      - *setup_monitoring
+
+  setup_fast_integration:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: fast_integration
```

Comment:
I see labels as in https://github.com/aeternity/infrastructure/blob/56cfd6326b59f8e91afe53397b7c1d2f5ae11eae/ansible/environments.yml#L113 ok